hermes/src/db/local_cache.rs

```diff
@@ -0,0 +1,88 @@
+use {
+    super::{
+        Db,
+        DbRecord,
+        RequestTime,
+    },
+    anyhow::Result,
+    dashmap::DashMap,
+    std::{
+        collections::VecDeque,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone)]
+pub struct LocalCache {
+    cache:            Arc<DashMap<Vec<u8>, VecDeque<DbRecord>>>,
+    max_size_per_key: usize,
+}
+
+impl LocalCache {
+    pub fn new(max_size_per_key: usize) -> Self {
+        Self {
+            cache: Arc::new(DashMap::new()),
+            max_size_per_key,
+        }
+    }
+}
+
+impl Db for LocalCache {
+    /// Add a new db record to the deque for the key while keeping it sorted.
+    /// This function also removes the oldest record in the cache if the max_size is reached.
+    /// Records are usually added in increasing order and the function is optimized for that case.
+    fn insert(&mut self, key: &[u8], record: DbRecord) -> Result<()> {
```

Comment:
imo you should keep `DbRecord` internal to the DB implementation. You have to query with a unix timestamp below anyway, so make insert take a timestamp and the value as arguments. Then the `Db` trait is something like a `TemporalKeyValueStore` or something.

Reply:
Agree, just to expand. It would be nice if once `Vec<u8>` is typed, that that is the return and insert type. That way we keep a typed interface to the database.